solidity/ecdsa/contracts/WalletRegistry.sol

```diff
@@ -0,0 +1,280 @@
+        dkg.setResultSubmissionEligibilityDelay(
+            newResultSubmissionEligibilityDelay
+        );
+    }
+
+    /// @notice Registers the caller in the sortition pool.
+    function registerOperator() external {
+        address operator = msg.sender;
+
+        require(
+            !sortitionPool.isOperatorInPool(operator),
+            "Operator is already registered"
+        );
+
+        sortitionPool.insertOperator(
+            operator,
+            staking.eligibleStake(operator, address(this))
+        );
+    }
+
+    /// @notice Updates the sortition pool status of the caller.
+    function updateOperatorStatus() external {
+        sortitionPool.updateOperatorStatus(
+            msg.sender,
+            staking.eligibleStake(msg.sender, address(this))
+        );
+    }
+
+    /// @notice Requests a new wallet creation.
+    /// @dev Can be called only by the owner of wallets.
+    ///      It starts a new DKG process.
+    function requestNewWallet() external {
+        require(msg.sender == walletOwner, "Caller is not the Wallet Owner");
+
+        dkg.lockState();
+        dkg.start(
+            uint256(keccak256(abi.encodePacked(relayEntry, block.number)))
+        );
+    }
+
+    /// @notice Submits result of DKG protocol. It is on-chain part of phase 14 of
+    ///         the protocol. The DKG result consists of result submitting member
+    ///         index, calculated group public key, bytes array of misbehaved
+    ///         members, concatenation of signatures from group members,
+    ///         indices of members corresponding to each signature and
+    ///         the list of group members.
+    ///         When the result is verified successfully it gets registered and
+    ///         waits for an approval. A result can be challenged to verify the
+    ///         members list corresponds to the expected set of members determined
+    ///         by the sortition pool.
+    ///         A candidate wallet is registered based on the submitted DKG result
+    ///         details.
+    /// @dev The message to be signed by each member is keccak256 hash of the
+    ///      calculated group public key, misbehaved members indices and DKG
+    ///      start block. The calculated hash should be prefixed with prefixed with
+    ///      `\x19Ethereum signed message:\n` before signing, so the message to
+    ///      sign is:
+    ///      `\x19Ethereum signed message:\n${keccak256(groupPubKey,misbehavedIndices,startBlock)}`
+    /// @param dkgResult DKG result.
+    function submitDkgResult(DKG.Result calldata dkgResult) external {
+        dkg.submitResult(dkgResult);
+    }
+
+    /// @notice Notifies about DKG timeout. Pays the sortition pool unlocking
+    ///         reward to the notifier.
+    function notifyDkgTimeout() external {
+        dkg.notifyTimeout();
+
+        // TODO: Implement transferDkgRewards
+        // transferDkgRewards(msg.sender, sortitionPoolUnlockingReward);
```

Comment:
Do we already know how will the reward be funded? It's not mentioned in https://github.com/keep-network/tbtc-v2/blob/main/docs/rfc/rfc-1.adoc#transaction-incentives.

Reply:
92a81ba02615791a30d69d533a0bd0571d192978